R/trackhub.R

```diff
@@ -0,0 +1,69 @@
+### =========================================================================
+### TrackHub support
+### -------------------------------------------------------------------------
+
+### - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+### TrackHub class
+###
+
+setClass("TrackHub", representation(uri = "character"))
+
+isFileReference <- function(s) {
+  length(grep(".txt", s)) == 1
+}
+
+uri <- function(x, ...) x@uri
+
+getContent <- function(x) {
+  content <- readLines(x, warn = FALSE)
+  rexp <- "^(\\w+)\\s?(.*)$"
+  data.frame(field = sub(rexp, "\\1", content),
+             value = sub(rexp,"\\2", content))
+}
+
+getValue <- function(field, content) {
+  position <- grep(field, content$field)
+  content$value[position]
+}
+
+hubFile <- function(x) file.path(uri(x), "hub.txt")
+
+genomesFile <- function(x, y) file.path(uri(x), y)
+
+setMethod("genome", "TrackHub", function(x) {
```

Comment:
Since we seem to be modeling the TrackHub as a list by genome, why not have this be a `names()` method? Then we'll get autocomplete for free, which will be useful as soon as we get support for e.g. `hub[["hg19"]]` or `hub$hg19`. Maybe it's even worth extending the `List` class from S4Vectors to get more freebies.